### DISCUSSION TITLE
Is it possible to add a chat with pdf feature?

### DISCUSSION THREAD

**Discussion by 10ANT** [LostRuins/koboldcpp#659] [Ideas]
for example, you have a book or a short journal document and you upload it to koboldcpp and based on the model your using it can give you incites on the pdf you upload and you can ask a question about it just and idea. Anyways, Thank you for your work it has helped me tremendously!

**Comment by LostRuins** [comment]
You would have to add it manually, either using the "Memory" or "World Info" feature.

**Reply by strikaco** [reply]
Does either use RAG or does it just flood the context with the entirety of the content?

**Comment by kalle07** [comment]
if you want it simple and good
use **pdfgear** its offline and only 300MB   ;)  and the bests including all github pdf/localdocs options like GPT4all and so on ...

**Reply by 10ANT** [reply]
Thank you so much!

**Reply by kalle07** [reply]
sry ... its not completely offline, i cant finde out which service is used (so yes GPT3.5) but i dont think that the whole document is uploaded

**Reply by dan-1am** [reply]
If your pdfs have embedded text, you could use pdftotext util from poppler/xpdf to convert them to text files.

**Reply by kalle07** [reply]
jan has made it with an llm and completely offline

https://github.com/janhq/jan
